src/vector/geometry.rs

```diff
@@ -341,6 +343,152 @@ impl Geometry {
             gdal_sys::OGR_G_AssignSpatialReference(self.c_geometry(), spatial_ref.to_c_hsrs())
         };
     }
+
+    /// Computes and returns the bounding envelope for this geometry
+    pub fn envelope(&self) -> gdal_sys::OGREnvelope {
+        let mut envelope = OGREnvelope {
+            MinX: 0.0,
+            MaxX: 0.0,
+            MinY: 0.0,
+            MaxY: 0.0,
+        };
+        unsafe { gdal_sys::OGR_G_GetEnvelope(self.c_geometry(), &mut envelope) };
+        envelope
+    }
+
+    /// Computes and returns the bounding envelope (3D) for this geometry
+    pub fn envelope3d(&self) -> gdal_sys::OGREnvelope3D {
+        let mut envelope = OGREnvelope3D {
+            MinX: 0.0,
+            MaxX: 0.0,
+            MinY: 0.0,
+            MaxY: 0.0,
+            MinZ: 0.0,
+            MaxZ: 0.0,
+        };
+        unsafe { gdal_sys::OGR_G_GetEnvelope3D(self.c_geometry(), &mut envelope) };
+        envelope
+    }
+
+    /// Determines whether two geometries intersect.
+    /// If GEOS is enabled, then this is done in rigorous fashion otherwise TRUE is returned if the envelopes (bounding boxes)
+    /// of the two geometries overlap.
+    pub fn intersects(&self, other: &Geometry) -> bool {
+        unsafe { gdal_sys::OGR_G_Intersects(self.c_geometry(), other.c_geometry()) == 1 }
+    }
+
+    /// Tests if this geometry and the other geometry are disjoint.
+    /// Geometry validity is not checked. In case you are unsure of the validity of the input geometries,
+    /// call [`is_valid()`] before, otherwise the result might be wrong.
+    ///
+    /// This function is built on the GEOS library, check it for the definition of the geometry operation.
+    /// If OGR is built without the GEOS library, this function will return an Error
+    pub fn disjoint(&self, other: &Geometry) -> Result<bool> {
+        unsafe { gdal_sys::CPLErrorReset() };
```

Comment:
Nit:

```suggestion
        unsafe { gdal_sys::CPLErrorReset(); }
```

(same for the others)

Reply:
This does not seems to follow the actual convention: (see for example cf https://github.com/georust/gdal/blob/master/src/vector/feature.rs#L218 and others, https://github.com/georust/gdal/blob/master/src/dataset.rs#L623, ...)

Reply:
Yeah, in that case no need to worry about it.